src/extractor/guidance/turn_handler.cpp

```diff
@@ -31,6 +51,67 @@ bool isEndOfRoad(const ConnectedRoad &,
            angularDeviation(possible_right_turn.angle, possible_left_turn.angle) >
                2 * NARROW_TURN_ANGLE;
 }
+
+template <typename Iterator>
+Iterator findOutermostForkCandidate(const Iterator start, const Iterator end)
+{
+    const auto outermost = std::adjacent_find(start, end, isOutermostForkCandidate);
+    if (outermost != end)
+    {
+        return outermost;
+    }
+    // if all roads are part of a fork, set `candidate` to the last road
+    else
+    {
+        return outermost - 1;
+    }
+}
+
+struct StraightestTurnAtIntersection
```

Comment:
After checking: given that it is used twice, I feel that introducing a dedicated struct here only adds confusion. Having the indirection using an `id` instead of the iterator `findClosestTurn` returns does only add confusion in my view.